test/sequential/test-dgram-bind-shared-ports.js

```diff
@@ -29,47 +29,52 @@ if (cluster.isMaster) {
   const worker1 = cluster.fork();
 
   if (common.isWindows) {
-    const checkErrType = (er) => {
+    const checkErrType = common.mustCall((er) => {
       assert.strictEqual(er.code, 'ENOTSUP');
       worker1.kill();
-    };
+    });
 
-    worker1.on('error', common.mustCall(checkErrType, 1));
+    worker1.on('error', common.mustCall(checkErrType));
     return;
   }
 
-  worker1.on('message', (msg) => {
+  worker1.on('message', common.mustCall((msg) => {
     assert.strictEqual(msg, 'success');
     const worker2 = cluster.fork();
 
-    worker2.on('message', (msg) => {
+    worker2.on('message', common.mustCall((msg) => {
       assert.strictEqual(msg, 'socket2:EADDRINUSE');
       worker1.kill();
       worker2.kill();
-    });
-  });
+    }));
+  }));
 } else {
-  const socket1 = dgram.createSocket('udp4', common.noop);
-  const socket2 = dgram.createSocket('udp4', common.noop);
+  const socket1 = dgram.createSocket('udp4', common.mustCall());
+  const socket2 = dgram.createSocket('udp4', common.mustCall());
+  socket1.ref();
+  socket2.ref();
 
-  socket1.on('error', (err) => {
-    // no errors expected
-    process.send(`socket1:${err.code}`);
-  });
+  socket1.on('error', common.mustNotCall());
 
-  socket2.on('error', (err) => {
+  socket2.on('error', common.mustCall((err) => {
     // an error is expected on the second worker
     process.send(`socket2:${err.code}`);
-  });
+    socket2.close(common.mustCall(() => {
```

Comment:
FWIW, I think this will only fire in the second worker. The first worker doesn't get an error on `socket2`.

Starting to wonder if whatever complexity is required to make `socket2.close()` work isn't any greater than complexity that might be required to make `port: 0` work for this test. :-|